src/components/SidebarState.vue

```diff
@@ -34,72 +80,125 @@ export default Vue.extend({
             // nextTick is needed because app_width is 0 on created()
             let sidebarDefault = this.$state.setting('sidebarDefault');
             if (sidebarDefault && this.$state.ui.app_width > 769) {
-                this.sidebarSection = sidebarDefault;
-                this.sidebarOpen = true;
+                this.channelState.sidebarSection = sidebarDefault;
+                this.channelState.sidebarOpen = true;
+                this.queryState.sidebarOpen = true;
             }
         });
     },
     methods: {
         section() {
+            if (this.sidebarOpen === null) {
+                this.sidebarOpen = this.channelState.sidebarOpen;
+            }
             if (this.isClosed) {
                 return '';
             }
 
-            let section = this.sidebarSection;
-            let isChannel = this.$state.getActiveBuffer().isChannel();
-
-            if (section === 'settings' && isChannel) {
-                return 'settings';
-            } else if (section === 'user' && this.sidebarUser && isChannel) {
-                return 'user';
-            } else if (section === 'nicklist' && isChannel) {
-                return 'nicklist';
-            } else if (section === 'about') {
-                return 'about';
+            let buffer = this.$state.getActiveBuffer();
+            if (buffer.isChannel()) {
+                if (
+                    this.sidebarOpen &&
+                    this.sidebarSection === 'user' &&
+                    !buffer.hasNick(this.sidebarUser.nick)
+                ) {
+                    // if the channel does not have the user show the nicklist
+                    this.showSidebar('nicklist');
+                }
+                return this.sidebarSection;
+            } else if (buffer.isQuery()) {
+                if (
+                    this.sidebarOpen &&
+                    (!this.sidebarUser || this.sidebarUser.nick !== buffer.name)
+                ) {
+                    let user = this.$state.getUser(buffer.getNetwork().id, buffer.name);
+                    // if the query sidebar is open and not the current user, switch users
+                    this.showSidebar('user', { user: user });
+                }
+                return this.sidebarSection;
             }
 
             return '';
         },
```

Comment:
This `section` method now has introduced side effects. It should simply return what section shoudl be displayed in the sidebar if it is open, however you now have it opening the sidebar in some cases

Reply:
This ensures that the sidebar shows the user for the current query. ie switching queries will always show the correct user